### ISSUE TITLE
AppDaemon, set_state, no customizations with 20171215?

### ISSUE THREAD

**Issue by randellhodges** [andrey-git/home-assistant-custom-ui#62]
### Checklist
- [X] I'm running the latest version of CustomUI ([Update guide](https://github.com/andrey-git/home-assistant-custom-ui/blob/master/docs/installing.md#updating)) or using a specific [release](https://github.com/andrey-git/home-assistant-custom-ui/releases) that is not marked as "Broken".
- [X] I tried to force-refresh (Ctrl+Shift+R / Ctrl+F5) the browser
- [X] (Optional, but recommended) I'm using Chrome or tried to reproduce the feature on Chrome.

**Browser + Version:** Chrome 63

**CustomUI version:** 20171215

**Home Assistant release (`hass --version`):** 0.60.0 (and previous version) 

**Problem-relevant `configuration.yaml` entries:**
```yaml
sensor.living_room_temp:
  friendly_name: 'Living Room Temperature'
  unit_of_measurement: '°F'
  group:
    group.group_livingroom:
      friendly_name: 'Temperature'
```

**Problem-relevant Home Assistant log entries:**
```
None
```

**Any errors from browser Javascript console:**
```
None
```
In some cases, the customizations are not being applied.

I have some WirelessTags sensors that does an HTTP post into AppDaemon with temperature, humidity and battery voltage. Because these things are chatty and have a limited api, I have them all into AppDaemon and do some cleanup.

In a nutshell, AppDaemon will get the existing state (if any), using get_state, then it will check for a difference (temperature comes in like 6 decimal places and I trunc to 2) and if it is different, it will then do a set_state.

This was working before the update to CustomUI that uses templates. I think the updates don't play well with me calling set_state.

Other customizations are applied to other entities, just not these.

**Comment by andrey-git** [comment]
Are you preserving all attributes between get_state and set_state?
Check the resulting attributes `/dev-state` panel. Do you see all customUI attributes like you configured them?

**Comment by randellhodges** [comment]
I wasn't before the change. I tried updating my app to try to preserve
them, but when I did a get_state all, I didn't get anything.

I am going to roll back the UI update to be 100% sure.

There are no attributes in the dev settings.

I will try to get better information either this evening or tomorrow
evening.

**Comment by randellhodges** [comment]
I rolled back to 20171129 and I have the same problems. I'm not sure what has changed, but it doesn't look like it is a CustomUI problem. Sorry about that.

**Comment by randellhodges** [comment]
If the only attributes are from the customization file, should I see them with a get_state api call?

**Comment by andrey-git** [comment]
Yes, get_state should return all attributes including those set via customization

**Comment by randellhodges** [comment]
Ok, I guess I need to keep digging. I run this in docker so I rolled back and tried 59.1, 59.2 and 60.0 with 20171129 and still not working. The AppDaemon get_state has an empty attributes dictionary.

Now to go triple check my customization file to make sure something isn't amiss there.

**Comment by randellhodges** [comment]
I gave up trying to figure out why it stopped working. I created a new component, a new sensor type called appdaemon that is basically a template sensor without a template. It has no state logic in it and relies solely on appdaemon to set it.

It seems to be working. My 10 minute "fix".

**Comment by andrey-git** [comment]
Marking as closed...
